src/foundry/client/documents/abstract/world-collection.d.mts

```diff
@@ -1,66 +1,63 @@
-
   /**
    * Reference the set of Folders which contain documents in this collection
    */
   get folders(): WorldCollection.Folders<DocumentName>;
 
   /**
    * Return a reference to the SidebarDirectory application for this WorldCollection.
-   * @remarks
-   * In the case where `Lowercase<Name>` is not a property of {@linkcode ui}, this actually always returns `undefined`,
-   * but {@linkcode RollTables} overrides this, so we need to allow a wider return type.
+   * @remarks This getter has several hardcoded special cases, see {@linkcode WorldCollection.Directory}
    */
-  get directory(): WorldCollection.Directory<DocumentName, Name>;
+  get directory(): WorldCollection.Directory<DocumentName>;
 
   /**
    * Return a reference to the singleton instance of this WorldCollection, or null if it has not yet been created.
    */
-  static get instance(): WorldCollection<Document.WorldType, any>; // TODO: Find a way to type this more concretely. One option would be to separate the static and non static side of this class, which allows accessing the the static this type to use the `documentName`.
+  // TODO: Find a way to type this more concretely. One option would be to separate the static and non static side of this class,
+  // TODO: which allows accessing the the static this type to use the `documentName`.
+  static get instance(): InitializedOn<WorldCollection.Any, "setup">;
 
-  // Note(LukeAbby): Due to the usage of `this["contents"]` in the parent class the override has
-  // to stay like this.
   protected override _getVisibleTreeContents(): this["contents"];
 
   /**
    * Import a Document from a Compendium collection, adding it to the current World.
    * @param pack       - The CompendiumCollection instance from which to import
    * @param id         - The ID of the compendium entry to import
-   * @param updateData - Optional additional data used to modify the imported Document before it is created
-   *                     (default: `{}`)
-   * @param options    - Optional arguments passed to the {@linkcode WorldCollection.fromCompendium} and {@linkcode Document.create} methods
-   *                     (default: `{}`)
+   * @param updateData - Optional additional data used to modify the imported Document before it is created (default: `{}`)
+   * @param options    - Optional arguments passed to the {@linkcode WorldCollection.fromCompendium | WorldCollection#fromCompendium} and
+   * {@linkcode Document.create} methods (default: `{}`)
    * @returns The imported Document instance
+   *
+   * @remarks The `updateData` parameter is {@link foundry.utils.mergeObject | merged} with the return of `WorldCollection#fromCompendium`,
+   * before being passed to `.create`, making the `DeepPartial<CreateData>` more correct than `UpdateData`.
+   *
+   * As noted in the parameter description, `options` is passed to both methods without alteration, and thus is typed as an intersection of
+   * the relevant interfaces.
```

Comment:
Well users may be confused about certain parameters that are completely useless. Like TypeScript will still allow a call like:
```ts
const options = { foo: 123, bar: 456 };
usesFoo(option);
usesBar(options);
```

But there's no reason why `usesFoo({ foo: 123, bar: 456 })` needs to be accepted, right?

I believe this may apply to an earlier interface too but GitHub changed how the review process works and will only show one file at at time when reviewing a big PR so I can't be bothered to check.